domain-models-runtime/src/main/java/org/folio/rest/persist/PreparedRowStream.java

```diff
@@ -75,8 +77,13 @@ public PreparedRowStream(PreparedStatement preparedStatement, int fetch, Tuple t
   }
 
   @Override
-  public void close() {
-    rowStream.close(close -> preparedStatement.close());
+  public Future<Void> close() {
+    Promise<Void> p = Promise.promise();
+    rowStream.close(close -> {
+      preparedStatement.close();
+      p.complete();
+    });
+    return p.future();
```

Comment:
This code doesn't propagates failures from preparedStatement.close() but it should. Simply use the existing close(Handler) code:

```
  @Override
  public Future<Void> close() {
    Promise<Void> promise = Promise.promise();
    close(promise);
    return promise.future();
  }
```

Reply:
done.